api/v1alpha3/awsmachine_webhook.go

```diff
@@ -78,14 +90,33 @@ func (r *AWSMachine) ValidateUpdate(old runtime.Object) error {
 	delete(oldAWSMachineSpec, "additionalSecurityGroups")
 	delete(newAWSMachineSpec, "additionalSecurityGroups")
 
+	// allow changes to secretARN
+	if cloudInit, ok := oldAWSMachineSpec["cloudInit"].(map[string]interface{}); ok {
+		delete(cloudInit, "secretARN")
+	}
+
+	if cloudInit, ok := newAWSMachineSpec["cloudInit"].(map[string]interface{}); ok {
+		delete(cloudInit, "secretARN")
+	}
+
 	if !reflect.DeepEqual(oldAWSMachineSpec, newAWSMachineSpec) {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "cannot be modified"))
-		return apierrors.NewInvalid(
-			GroupVersion.WithKind("AWSMachine").GroupKind(),
-			r.Name, allErrs)
 	}
 
-	return nil
+	return aggregateObjErrors(r.GroupVersionKind().GroupKind(), r.Name, allErrs)
+}
+
+func (r *AWSMachine) validateCloudInitSecret() field.ErrorList {
+	// if cloudInit was already defined only allow mutation of secretARN.
+	// else we allow a new set of cloudInit options
```

Comment:
Can remove this comment or reword to say that secret arn is disallowed if encrypted is false